esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -99,44 +98,40 @@ IPropertyTree * fetchESDLDefinitionFromDali(const char *esdlServiceName, unsigne
     StringBuffer id(esdlServiceName);
     id.toLowerCase().append(".").append(targetVersion);
 
-    return fetchESDLDefinitionFromDaliById(id);
+    return fetchESDLDefinitionFromDaliById(id, exsdl);
 }
 
 IPropertyTree * fetchESDLBindingFromDali(const char *process, const char *bindingName)
 {
-    Owned<IRemoteConnection> conn;
     try
     {
-        conn.set(querySDS().connect(ESDL_BINDINGS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL));
+        //There shouldn't be multiple entries here, but if so, we'll use the first one
+        VStringBuffer xpath("%s[@id='%s.%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
+
+        Owned<IRemoteConnection> conn = querySDS().connect(ESDL_BINDINGS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
         if (!conn)
         {
-            DBGLOG("Unable to connect to ESDL Service binding information in dali %s", ESDL_BINDINGS_ROOT_PATH);
-            return NULL;
+            ESPLOG(LogMin, "Unable to connect to ESDL Service binding information in dali %s", ESDL_BINDINGS_ROOT_PATH);
+            return nullptr;
         }
 
-        conn->close(false); //release lock right away
-
-        DBGLOG("ESDL Binding: Fetching ESDL Binding from Dali %s[@EspProcess='%s'][@EspBinding='%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
+        ESPLOG(LogNormal, "ESDL Binding: Fetching ESDL Binding from Dali %s[@EspProcess='%s'][@EspBinding='%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
 
         IPropertyTree *esdlBindings = conn->queryRoot();
         if (!esdlBindings)
         {
-            DBGLOG("Unable to open ESDL Service binding information in dali %s", ESDL_BINDINGS_ROOT_PATH);
-            return NULL;
+            ESPLOG(LogMin, "Unable to open ESDL Service binding information in dali %s", ESDL_BINDINGS_ROOT_PATH);
+            return nullptr;
         }
 
-        //There shouldn't be multiple entries here, but if so, we'll use the first one
-        VStringBuffer xpath("%s[@id='%s.%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
-
-        return esdlBindings->getPropTree(xpath);
+        IPropertyTree  * esdlBinding = createPTreeFromIPT(esdlBindings->queryPropTree(xpath));
+        return esdlBinding;
     }
     catch(...)
     {
-        DBGLOG("ESDL Binding: Unknown error encountered while fetching ESDL Binding from Dali %s[@EspProcess='%s'][@EspBinding='%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
-        if (conn)
-            conn->close(false); //release lock right away
-
+        ESPLOG(LogMin, "ESDL Binding: Unknown error encountered while fetching ESDL Binding from Dali %s[@EspProcess='%s'][@EspBinding='%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
```

Comment:
minor: Not new, but we should really be tracing any exception and everything should be an IException.